docs/content/v0.20/release.md

```diff
@@ -0,0 +1,60 @@
+- A user can pick up and use a given existing release without being perturbed by on-going contributor work. A release is an immutable thing.
+- A release with a given semver identifier is built from a commit of this Git repository tagged with a tag whose name is "v" followed by the release identifier.
+- There is a concept of a release branch, with a name following the pattern `release-$major.$minor`. This allows contributors to do maintenance work on older release branches. This also allows a user to track work along such a line.
+- The contents of `main` always work. This includes passing CI tests. This includes documentation being accurate. We allow point-in-time specific documentation, such as a document that says "Here is how to use release 1.2.3" --- which would refer to a release made in the past. We do not require the documentation in `main` to document all releases.
+- A git tag is immutable. Once associated with a given Git commit, that association is not changed later.
+- We do not put self-references into Git. For example, making release `1.2.3` doe not require changing any file in Git to have the string `1.2.3` in it.
+
+We have the following limitations.
+
+- The only way to publish artifacts (broadly construed, not (necessarily) GitHub "release artifacts") is to make a release.
+- The only way to test published artifacts is to make a release and test it.
+- Thus, it is necessary to keep users clearly appraised of the quality (or status of evaluating the quality) of each release.
+- Because of the lack of self references, most instructions and tests do not have concrete release identifiers in them; instead, the user has to chose and supply the release identifier. There can also be documentation of a specific past release (e.g., the latest stable release) that uses the literal identifier for that past release.
+- Because of the prohibition of self references, Git will not contain the exact bytes of our Helm chart definitions. Where a Helm chart states its own version or supplies a default value for a container image reference to an image built from the same release, the bytes in Git have a placeholder for that image's tag and the process of creating the published release artifacts fills in that placeholder. Think of this as being analogous to the linking done when building a binary executable file.
+
+## Technology
+
+There will be a GitHub workflow that creates the published artifacts for each Git tag whose name starts with "v". The rest of the tag name is required to be a semver release identifier. For each tag `v$version` the following published artifacts will be created.
+
+- The container image for the kubestellar-controller-manager (KCM), at `ghcr.io/kubestellar/kubestellar-controller-manager` (or should it be `ghcr.io/kubestellar/kubestellar/controller-manager` from the ks/ks repo? or just `ghcr.io/kubestellar/controller-manager`?). Image tag will be `$version`.
```

Comment:
I realize now that our naming scheme for packages published to ghcr.io is quite inconsistent. Some package has the scheme `ghcr.io/<org>/<repo>/<name>` and some  `ghcr.io/<org>/<name>`. It looks `ghcr.io/<org>/<repo>/<name>` may be better as we could have for example `ghcr.io/kubestellar/kubestellar/controller-manager` and `ghcr.io/kubestellar/ocm-status-addon/controller-manager`.

Reply:
IMO the names should be following the pattern `ghcr.io/kubestellar/{package_name}`. 
I do not see the point in duplicating kubestellar identifier.

I don't think we should re-invent the wheel. 
controller manager is an abstract term. from a quick search of CNCF projects images I haven't found any project that uses such an abstract name for image (I've checked 5-6 projects, not the whole list). 
We should use specific package names instead, e.g., `binding-controller`, `ocm-transport-plugin-controller`, `ocm-status-addon-controller`, etc. 
there are many examples for how CNCF projects are naming their images, most of them are using the pattern of `{Registry}/{Org}/{Package}`. 
 
some example for such are:
`quay.io/coreos/etcd-operator`
`quay.io/open-cluster-management/registration-operator`.
`openpolicyagent/opa-kube-scheduler` (in docker hub)
and more..

Reply:
@nirrozenbaum there is more than one controller within the kubestellar controller manager, that is why is a controller manager. An example of that is the kube-controller-manager in Kubernetes. Just calling it binding controller may be inaccurate (it includes also status controller). Using `ghcr.io/kubestellar/{package-name}` is fine with me as long as we have distinct and meaningful names. Possible choices are `ghcr.io/kubestellar/ks-controller-manager`, or `ghcr.io/kubestellar/kubestellar-controller-manager`. If we don't include the name of the repo in the path (`ghcr.io/kubestellar/<repo-name>/`) I think we should at least have the repo name or some abbreviated form of the repo name in the package name.

Reply:
@nirrozenbaum why not including the repo ?  I think we should treat our repos as independent, and therefore the repo "owner" can use any package name. In that case we must have repo in order to avoid collisions

Reply:
@pdettori yes, using meaningful names was my intention. 

there is no problem with any of the approaches, I just stated that:
(a) no need for duplicate identifier
(b) the CNCF projects I checked are using this pattern. 

@ezrasilvera I actually think package names in this org should be validated, and not “repo owner can choose any name”. this is important for consistency.

eventually the difference will be:
ghcr.io/kubestellar/kubestellar-controller-manager Vs ghcr.io/kubestellar/kubestellar/controller-manager. 
which is better? 
there is no one correct answer but the other CNCF projects are mostly using the former.

Reply:
Note that the question of how a use refers to an OCI image is connected to a bit of GitHub structure, whether a package is "from" a repo or "from" the org.

It makes more sense to me for each repo to "own" its GitHub packages. I could live with packages owned by the org (necessarily with coordinated names).

Reply:
not sure I got the point about the listing in the browser UI. I used the pattern I suggested and I see in the packages that my package is from my repo. 
I've attached screenshot of packages list and if I go into that package I also see it's linked to my repo.

can you please explain the point you were mentioning? 

![image](https://github.com/kubestellar/kubestellar/assets/19717747/c745f435-ab2d-42b4-976a-e177bb2e34a1)

Reply:
@nirrozenbaum : my point is about the difference between the pattern you followed and the pattern followed by, e.g., the kubestellar-operator-chart. See how they are described differently?  And on the browser page for https://github.com/kubestellar/kubestellar/, you see that it says that repo has no packages.

Reply:
is this a limitation? 
I'm almost sure if you go into the package settings there is an option to link the package to the repo (should be done only once and will be kept for all releases)

Reply:
https://docs.github.com/en/packages/learn-github-packages/connecting-a-repository-to-a-package#connecting-a-repository-to-an-organization-scoped-package-on-github

Reply:
Thanks for the pointer. BTW, I did not know any way to tell GitHub that a particular package is "from" or "in" a particular repo. Are there any other ways? Maybe part of the process of publishing the package? @nirrozenbaum , @pdettori : how did you do it, in the cases where you did?

Reply:
the initial image I created was done manually and wasn’t linked. I linked it to the repo as described in the pointer doc.
later I removed the manually created images and created through go releaser and link was already there. 

I’m sure there is way to have as part of the automated process. maybe it is already there, I’m not sure cause in my case I created the link manually before.